components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -5355,14 +5363,8 @@ private boolean isImmutableClaim(String claim) throws UserStoreException {
         Map<String, String> claimMappings = SCIMCommonUtils.getSCIMtoLocalMappings();
 
         return claim.equals(claimMappings.get(SCIMConstants.CommonSchemaConstants.ID_URI)) ||
-                claim.equals(claimMappings.get(SCIMConstants.UserSchemaConstants.USER_NAME_URI)) ||
-                claim.equals(claimMappings.get(SCIMConstants.UserSchemaConstants.ROLES_URI + "."
-                        + SCIMConstants.DEFAULT)) ||
                 claim.equals(claimMappings.get(SCIMConstants.CommonSchemaConstants.CREATED_URI)) ||
-                claim.equals(claimMappings.get(SCIMConstants.CommonSchemaConstants.LAST_MODIFIED_URI)) ||
-                claim.equals(claimMappings.get(SCIMConstants.CommonSchemaConstants.LOCATION_URI)) ||
                 claim.equals(claimMappings.get(SCIMConstants.UserSchemaConstants.FAMILY_NAME_URI)) ||
-                claim.equals(claimMappings.get(SCIMConstants.UserSchemaConstants.GROUP_URI)) ||
                 claim.contains(UserCoreConstants.ClaimTypeURIs.IDENTITY_CLAIM_URI_PREFIX);
     }
```

Comment:
This logic is executed before deleting claims. It doesn't allow deleting local claims which are immutable. 

There set of local claims removed from the updating user and existing user. The claim difference are calculated based on that.

From the updating user the required immutable claims are correctly removed, but from the existing user, the immutable claims are not deleted. This PR has fixed that issue. 

Hence the immutable claims will not be detected as a claim difference for update, and the private method not required those claims also. 

The main issue was some immutable claims are detected as claim difference due to above mentioned issue, which was corrected by removing the immutable claims from the old claim list.